tests/test_main.py

```diff
@@ -0,0 +1,265 @@
+        state.change_labels.assert_called_once_with(LabelEvent.TRY)
+
+    @patch('homu.main.PullReqState')
+    def test_try_negative(self, MockPullReqState):
+        state = MockPullReqState()
+        _try(state, 'try-', True)
+        self.assertFalse(state.try_)
+        state.init_build_res.assert_called_once_with([])
+        state.save.assert_called_once()
+        assert not state.change_labels.called, 'change_labels was called and should never be.'
+
+    @patch('homu.main.PullReqState')
+    def test_clean(self, MockPullReqState):
+        state = MockPullReqState()
+        clean(state)
+        self.assertEqual(state.merge_sha, '')
+        state.init_build_res.assert_called_once_with([])
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_retry_try(self, MockPullReqState):
+        state = MockPullReqState()
+        state.try_ = True
+        retry(state, True)
+        state.set_status.assert_called_once_with('')
+        state.change_labels.assert_called_once_with(LabelEvent.TRY)
+
+    @patch('homu.main.PullReqState')
+    def test_retry_approved(self, MockPullReqState):
+        state = MockPullReqState()
+        state.try_ = False
+        retry(state, True)
+        state.set_status.assert_called_once_with('')
+        state.change_labels.assert_called_once_with(LabelEvent.APPROVED)
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_negative(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_negative(state)
+        self.assertEqual(state.delegate, '')
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_positive(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_positive(state, True)
+        state.get_repo.assert_called_once()
+        state.add_comment.assert_called_once()
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_to(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_to(state, True, 'delegate=user')
+        self.assertEqual(state.delegate, 'user')
+        state.save.assert_called_once()
+        state.add_comment.assert_called_once_with(
+            ':v: @user can now approve this pull request'
+        )
+
+    @patch('homu.main.PullReqState')
+    def test_set_priority_not_priority_less_than_max_priority(self, MockPullReqState):
+        state = MockPullReqState()
+        set_priority(state, True, 'p=1', {'max_priority': 3})
+        self.assertEqual(state.priority, 1)
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_set_priority_not_priority_more_than_max_priority(self, MockPullReqState):
+        state = MockPullReqState()
+        set_priority(state, True, 'p=5', {'max_priority': 3})
+        state.add_comment.assert_called_once_with(':stop_sign: Priority higher than 3 is ignored.')
```

Comment:
Add an assert that state.save wasn't called, the state's priority wasn't updated, and set_priority returned False?